client/src/components/ChatRoom.svelte

```diff
@@ -0,0 +1,161 @@
+<script lang="ts">
+  import { onMount, onDestroy } from "svelte";
+  import { authStore } from "$stores/auth";
+  import { onSnapshot, QueryDocumentSnapshot } from "firebase/firestore";
+  import {
+    chatMessageValidator,
+    type ChatMessage,
+    type ChatRoom,
+    type Lobby,
+  } from "$lib/firebase/firestore-types/lobby";
+  import { findChatRoom } from "$lib/firebase/chat";
+  import { getChatRoomMessagesCollection } from "$lib/firebase/firestore-collections";
+  import type { User } from "firebase/auth";
+  import type { UserData } from "../../../functions/src/firestore-types/users";
+  import { addChatMessage, deleteChatRooms } from "$lib/firebase/firestore-functions";
+  // props
+  export let lobbyData: Lobby & { id: string };
+  // variables
+  let user = $authStore as User;
+  let userInfo: UserData;
+  let partnerInfo: UserData;
+  let chatRoomInfo: QueryDocumentSnapshot<ChatRoom>;
+  let chatMessages: ChatMessage[] = [];
+  let message: string = "";
+  let timer: ReturnType<typeof setInterval>;
+  let countdown = 60;
+  let error = {
+    status: false,
+    message: "",
+  };
+
+  onMount(async () => {
+    // Query for their chatroom
+    chatRoomInfo = await findChatRoom(lobbyData.id, user.uid);
+    // subscribe the chat messages
+    onSnapshot(getChatRoomMessagesCollection(lobbyData.id, chatRoomInfo.id), (collection) => {
+      chatMessages = collection.docs.map((doc) => doc.data()).sort((a, b) => a.timestamp.seconds - b.timestamp.seconds);
+    });
+    // Get userInfo
+    userInfo = lobbyData.players[lobbyData.uids.indexOf(user.uid)];
+    // Get partnerInfo
+    const [partner] = chatRoomInfo.data().pair.filter((u) => {
+      return user.uid !== u;
+    });
+    partnerInfo = lobbyData.players[lobbyData.uids.indexOf(partner)];
+    // create timer
+    timer = setInterval(() => {
+      countdown -= 1;
+    }, 1000);
```

Comment:
this is not a reliable way to do a countdown.
if you're interested here is an exploration into javascript timer techniqes https://www.youtube.com/watch?v=MCi6AZMkxcU